# [LAB 07] 6. 일원분산분석
## #연습문제

In [1]:
from hossam import load_data
from helpers import *
from pandas import pivot_table

# 분산분석을 위한 라이브러리
from scipy.stats import f_oneway
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm
from pingouin import anova
from pingouin import welch_anova

📦 아이티윌 이광호 강사가 제작한 라이브러리를 사용중입니다.
📚 자세한 사용 방법은 https://py.hossam.kr 을 참고하세요.
📧 Email: leekh4232@gmail.com
🎬 Youtube: https://www.youtube.com/@hossam-codingclub
📝 Blog: https://blog.hossam.kr/
🔖 Version: 0.5.19


#### 문제 1 - 세가지 읽는 방법에 통계적으로 유의한 차이가 있는지 검정하라

In [4]:
df1 = load_data('reading')
df1

📚 유아들을 대상으로 세 가지 읽는 방법을 비교,실험하여 얻은 독서평가 점수 데이터 (출처: 방송통신대학교 통계학개론)


,독서방법,점수
0,B,7.700
1,B,1.500
2,B,4.300
3,B,5.400
4,B,4.300
5,B,0.700
6,B,4.100
7,B,8.900
8,B,8.100
9,B,9.200


In [3]:
my_stats.anova_oneway(df1, y='점수', between='독서방법')

,test,Source,ddof1,ddof2,F,p_unc,np2,effect_size
0,welch_anova,독서방법,2,48.204,10.787,0.000,0.283,Large


- 검정 선택 : test=welch_anova → 등분산성을 위반하여 Welch ANOVA 검정으로 진행
- 유의성 : F(2, 48.204)=10.787, p=0.000<0.05 → 귀무가설 기각, 세 독서방법 중 적어도 한 방법의 평균 점수는 다르다.
- 효과 크기 : np2=0.283 → Large, 독서방법 차이가 점수 변동의 약 28%를 설명, 통계적 유의성을 넘어 실질적으로도 큰 효과를 보였다.
- 다음 단계 : ANOVA는 '어딘가 차이가 있다'까지만 말해주므로, 어느 독서방법 쌍이 독서평가 점수가 다른지는 사후검정 등으로 확인 필요

#### 문제 2- 상권 종류별 2023년 한해 동안의 평균 매출액은 어떤 차이를 보이는지 분석하라

In [8]:
df2 = load_data('commercial')
df2.head()

📚 서울시의 2023년 상권에 따른 분기별 매출액과 매출건수에 대한 자료 (출처: 서울 열린데이터 광장)


,quarter,code,type,name,sales_amount,number_of_sales
0,1,3001491,관광특구,이태원 관광특구,72523644100,2496865
1,1,3001492,관광특구,명동 남대문 북창동 다동 무교동 관광특구,283332346393,10246122
2,1,3001493,관광특구,동대문패션타운 관광특구,81696730221,2880324
3,1,3001494,관광특구,종로·청계 관광특구,212062656625,4960006
4,1,3001495,관광특구,잠실 관광특구,231338386876,5995166


In [10]:
my_stats.anova_oneway(df2, y='sales_amount', between='type')

,test,Source,ddof1,ddof2,F,p_unc,np2,effect_size
0,welch_anova,type,3,110.239,161.433,0.000,0.228,Large


- 검정 선택 : test=welch_anova → 등분산성을 위반하여 Welch ANOVA 검정으로 진행
- 유의성 : F(3, 110.239)=161.433, p=0.000<0.05 → 귀무가설 기각, 네 상권 중 적어도 한 상권의 평균 매출액은 다르다.
- 효과 크기 : np2=0.228 → Large, 상권종류별 2023년 한해 동안의 평균매출액 차이가 약 22.8%를 설명, 통계적 유의성을 넘어 실질적으로도 큰 차이를 보였다.
- 다음 단계 : ANOVA는 '어딘가 차이가 있다'까지만 말해주므로, 어느 한 상권의 매출평균이 다른지는 사후검정 등으로 확인 필요